test/stdlib/StringDescribing.swift

```diff
@@ -22,4 +22,19 @@ StringDescribingTestSuite.test("String(reflecting:) should include extra stuff i
   expectEqual(privateName.suffix(5), ").Foo")
 }
 
+StringDescribingTestSuite.test("String(describing:) shouldn't include local discriminator") {
+  class LocalType {}
+  let t = LocalType()
+  expectEqual(String(describing: type(of: t)), "LocalType")
+}
+
+StringDescribingTestSuite.test("String(reflecting:) should uniquely qualify type name") {
+  class LocalType {}
+  let t = LocalType()
+  let representation = String(reflecting: type(of: t))
+
+  expectEqual(representation.prefix(6), "main.(")
+  expectEqual(representation.suffix(11), ").LocalType")
```

Comment:
Ah, now we should be checking that the "#1" *is* there. Doesn't this fail?

Reply:
It doesn't fail. Isn't it expected behaviour for locally defined types?

Reply:
Hm, no. If `String(reflecting:)` uniquely identifies a type, that means it has to include the "`#1`". Maybe look for where else ShowPrivateDiscriminators is set to true? Or just combine the two settings, since I can't think why we'd ever want them to be different.

Reply:
It looks like after merging [this](https://github.com/apple/swift/pull/13468/files#diff-360e359aecc8c097fac060c301c6d378R1005) change, the logic should be reconsidered.

Reply:
I don't think that affects this.

Reply:
Added a comment to the ticket.